### ISSUE TITLE
TensorRT engine build failure due to shape mismatch

### ISSUE THREAD

**Issue by MengX1991** [NVIDIA/TensorRT#2413]
## Description

We have an onnx model with cnn operator, we need to clean some values at certain width of the conv output based on a input mask. For example, if the conv result shape is nchw, the mask will be a boolen tensor with shape w. We're using a where operator to take mask as condition and take conv output as input. Since we have several maxpool between cnn operator, the mask should be shrink as well, currently, we use a slice operator to shrink the mask.

![image](https://user-images.githubusercontent.com/93172434/197466456-5cf378d4-3594-4b70-b5b7-3f7edfc8fef5.png)
The Slice steps and maxpool stride are both 2.

Previous we uses ONNXRuntime with cuda ep, and the solution could work well and the result is correct, recently, we want to move to TensorRT ep, but when we generate the engine, there is a shape mismatch failure.
![image](https://user-images.githubusercontent.com/93172434/197466394-c609664f-3d70-4af7-a8d7-d521712fb658.png)

If the min/max shape is not specified, the bug will not repro.

## Environment

**TensorRT Version**: 8.2.3
**NVIDIA GPU**: Tesla V100-PCIE-16GB
**NVIDIA Driver Version**: 460.32.03
**CUDA Version**: 11.2
**CUDNN Version**: 
**Operating System**: Linux
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## Steps To Reproduce

We can repro this bug both from onnxruntime and trtexec
/usr/src/tensorrt/bin/trtexec --onnx=./debug.onnx --saveEngine=./test --optShapes=conv1_relu:1x16x64x320,seq_mask:320  --minShapes=conv1_relu:1x16x64x16,seq_mask:16 --maxShapes=conv1_relu:1x16x64x640,seq_mask:640

**Comment by MengX1991** [comment]
Here is the model file we are using https://file.io/YW5dJgs2XiiW

**Comment by zerollzeng** [comment]
Looks like this has been fix in TRT 8.5, can you upgrade your TensorRT version?
```
&&&& PASSED TensorRT.trtexec [TensorRT v8501] # trtexec --onnx=./debug.onnx --saveEngine=./test --optShapes=conv1_relu:1x16x64x320,seq_mask:320 --minShapes=conv1_relu:1x16x64x16,seq_mask:16 --maxShapes=conv1_relu:1x16x64x640,seq_mask:640
```

**Comment by MengX1991** [comment]
> Looks like this has been fix in TRT 8.5, can you upgrade your TensorRT version?
> 
> ```
> &&&& PASSED TensorRT.trtexec [TensorRT v8501] # trtexec --onnx=./debug.onnx --saveEngine=./test --optShapes=conv1_relu:1x16x64x320,seq_mask:320 --minShapes=conv1_relu:1x16x64x16,seq_mask:16 --maxShapes=conv1_relu:1x16x64x640,seq_mask:640
> ```

Hi zerollzeng, thanks for your quick response and kind help, it will be great if this has already been fixed. Could you please point me where I can try the 8.5 version TRT? I only find 8.4.x in this repo or https://developer.nvidia.com/nvidia-tensorrt-8x-download.

**Comment by zerollzeng** [comment]
Sorry for the convenient, it should be released soon, for now you can try the EA with our docker image 22.09

**Comment by MengX1991** [comment]
Thanks @zerollzeng It works well on TRT 8.5. Do you know when we can get this release?

**Comment by nvpohanh** [comment]
TRT 8.5 GA has been released. Could we close this?

**Comment by zerollzeng** [comment]
close
